app/services/catalog/create_approval_request.rb

```diff
@@ -0,0 +1,62 @@
+module Catalog
+  class CreateApprovalRequest
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+    end
+
+    def process
+      @order.order_items.each do |order_item|
+        approvals = send_approval_requests(order_item)
+        approvals.compact.each do |req|
+          order_item.approval_requests << create_approval_request(req)
+        end
+      end
+
+      @order.update(:state => "Ordered", :ordered_at => Time.now.utc)
+      self
+    rescue Catalog::ApprovalError => e
+      Rails.logger.error("Error putting in approval Request for #{order.id}: #{e.message}")
+      raise
+    end
+
+    private
+
+    def send_approval_requests(order_item)
+      workflows(order_item.portfolio_item).each_with_object([]) do |workflow, approvals|
+        Approval.call do |api_instance|
+          approvals << api_instance.create_request(workflow, request_body_from(order_item))
+        end
+        order_item.update_message("info", "Approval Request Submitted for workflow #{workflow}")
```

Comment:
@lindgrenj6 I think this message should include the approval request id so we can correlate the messages coming from the listener.

Reply:
Done!